### ISSUE TITLE
Plugin feedback platform

### ISSUE THREAD

**Issue by philpax** [goatcorp/DIPs#34]
Our current mechanism for feedback is the #feedback channel, and while it does well for what it is, what it is has some problems:

- it's impossible to reach out to people who don't leave a contact method
- it's impossible to reach out to people who leave invalid/inaccessible content method (Discord IDs while not in the Goat Place server)
- it requires developers have a email address they're willing to reveal for people who contact by email
- unless you have the good fortune of a Discord user on Goat Place, it's difficult to have a synchronous conversation with them to diagnose the issue
- if you _do_ have a Discord user on Goat Place, the developer must now speak to them directly, and break "the wall" between the developer and the user, which can be awkward at best and downright calamitous at worst (stalkers exist...)

What I'd like to suggest, instead, is a feedback platform similar to a ticketing system. A user can hit the feedback button in-game, which will allow them to open a ticket that anyone with publishing / support rights to the plugin can respond to, in their own time. This ticket will remain open for a week after the last contact on either side, at which point it will be automatically closed.

The user side will look like a back-and-forth chat in the feedback system in-game. The support agent side will have a website where they can log in with their GH credentials, and they're allowed to interact with any tickets that their GH account has access to (can go with "has contributed to that repo", but I think we'd best make it explicit).

This is -probably- not too hard to implement, but I'm probably missing some complexity. I bet some of our support agents have thoughts on this...

**Comment by MidoriKami** [comment]
Might also be worth mentioning that if someone has third party repos installed, or if the specific plugin they are leaving feedback for is from a third party repo, display a disclaimer warning them that they will not be receiving support in the case of an issue.

A significant portion of feedback is regarding some kind of issue or bug, or is the user not understand the scope of the plugin.

**Comment by philpax** [comment]
It also occurs to me that we could potentially be cute and build a general-purpose plugin platform that could accommodate #15 as well. You log into the site with your credentials of your choice, and you can leave feedback for plugins or contribute a preset, etc

**Comment by philpax** [comment]
It could! We could roll the suggestions/feedback/presets into one website.

**Comment by PrincessRTFM** [comment]
> A significant portion of feedback is regarding some kind of issue or bug, or is the user not understand the scope of the plugin.

It might be reasonable (and should be fairly easy to implement, as a field in the plugin manifest) to allow plugins to display custom messages written by the developers directly on the feedback window, as well. The simplest would be a simple string field for a plaintext message, although I know there was talk of using a two-field system to allow including one or more image URLs.

**Comment by philpax** [comment]
@karashiiro is working on a preset sharing platform. It is possible that could be grown to accommodate this.

**Comment by karashiiro** [comment]
Yes, I'm working on that [here](https://github.com/karashiiro/plugin-platform) for now, and I'll migrate it over and write up a contributing guide once the preset-only version of it is mostly implemented. Feel free to assign me to this for the time being.
